### ISSUE TITLE
Steam client does not open: /lib/i386-linux-gnu/libpthread.so.0: cannot read file data

### ISSUE THREAD

**Issue by phausamann** [ValveSoftware/steam-for-linux#10604]
#### Your system information

* Steam client version (build number or date): 1.0.0.79
* Distribution (e.g. Ubuntu): ubuntu 20.04
* Opted into Steam client beta?: no
* Have you checked for system updates?: yes
* GPU: Nvidia RTX 3080 Mobile, driver version 550.54.14

#### Please describe your issue in as much detail as possible:

When launching the latest steam client, the execution fails with the following error:

```
~/.local/share/Steam/ubuntu12_32/steam: error while loading shared libraries: /lib/i386-linux-gnu/libpthread.so.0: cannot read file data
```

Full console output:

```
steam.sh[23712]: Running Steam on ubuntu 20.04 64-bit
steam.sh[23712]: STEAM_RUNTIME is enabled automatically
setup.sh[23841]: Steam runtime environment up-to-date!
~/.local/share/Steam/ubuntu12_32/steam-runtime/run.sh: line 85: steam-runtime-identify-library-abi: command not found
run.sh[23853]: steam-runtime-identify-library-abi --ldconfig-paths failed, falling back to ldconfig
steam.sh[23712]: Error: You are missing the following 32-bit libraries, and Steam may not run:
libc.so.6
steam.sh[23712]: Can't find 'steam-runtime-check-requirements', continuing anyway
~/.local/share/Steam/ubuntu12_32/steam: error while loading shared libraries: /lib/i386-linux-gnu/libpthread.so.0: cannot read file data
```

#### Steps for reproducing this issue:

1. Download the debian package with the latest steam client
2. Install with `sudo apt install <path/to/steam_latest.deb>`
3. Launch `steam`

I have already tried the following troubleshooting steps:

* Ran `steam --reset`
* Did a reinstall of the steam client
* Installed required 32 bit system libraries: `sudo apt install libgl1-mesa-dri:i386 libgl1-mesa-glx:i386 libc6:i386`
* Re-installed `libc6` which contains `libpthread`: `sudo apt install --reinstall libc6:i38`
* Checked that `libpthread` is not corrupted by running `nm -D /lib/i386-linux-gnu/libpthread.so.0`

**Comment by kisak-valve** [comment]
Hello @phausamann, this reads like a failing hard drive or damaged filesystem. Can you check that your system drive and filesystem are healthy? Also, please check of any 32 bit application works on your system.

**Comment by ericrwinkler** [comment]
I'm seeing this error as well when I run try to run steam vr through Reaper in the command line. I've got several boxes where I do this and all of them have the same issue either complaining about libpthread of libc.6.

**Comment by sammy-net** [comment]
I'm seeing a similar issue.  Steam client 1.0.0.79, Ubuntu 22.04.
```
steam.sh[2475833]: Running Steam on ubuntu 22.04 64-bit
steam.sh[2475833]: STEAM_RUNTIME is enabled automatically
setup.sh[2478272]: Steam runtime environment up-to-date!
steam.sh[2475833]: Error: You are missing the following 32-bit libraries, and Steam may not run:
libc.so.6
steam.sh[2475833]: Steam client's requirements are satisfied
/home/sammy/.steam/debian-installation/ubuntu12_32/steam: error while loading shared libraries: /lib/i386-linux-gnu/libc.so.6: cannot read file data
```

**Comment by phausamann** [comment]
Seems to be fixed with the latest update, build number 1709846872
